rustls/src/error.rs

```diff
@@ -262,6 +262,19 @@ pub enum CertificateError {
     Other(Arc<dyn StdError + Send + Sync>),
 }
 
+impl From<CertificateError> for AlertDescription {
+    fn from(e: CertificateError) -> Self {
+        use CertificateError::*;
+        match e {
+            BadEncoding | UnhandledCriticalExtension | NotValidForName => Self::BadCertificate,
+            Expired | NotValidYet => Self::CertificateExpired,
+            UnknownIssuer => Self::UnknownCA,
+            BadSignature => Self::DecryptError,
+            Other(_) => Self::CertificateUnknown,
```

Comment:
`CertificateUnknown` seems like a surprising choice for `Other`. Maybe this should also become `BadCertificate`? We're already in the context of `CertificateError`s so that seems like a better choice to me.

I guess this is in line with OpenSSL and BoringSSL.

Reply:
Following the [RFC](https://www.rfc-editor.org/rfc/rfc5246#section-7.2.2)

>    certificate_unknown
      Some other (unspecified) issue arose in processing the
      certificate, rendering it unacceptable.

But it's just my own understand.
Should we really following other implementation ?

Reply:
Okay, good point. Maybe add these RFC references to the code as well?

Reply:
@zkonge It looks like 77199400810439fb1d99654de73654facbb2f457 added the OpenSSL/BoringSSL references in-code, but left the RFC references just to the commit message. Was that intentional?

Reply:
It's intentional. Because RFC doesn't tell us how to mapping X.509 certificate error into TLS alerts.
But thanks, you remind me that I forgot to add comments for puzzling mapping choices.